File: src/main/java/ninekothecat/catconomy/defaultImplementations/CatPrefix.java
package ninekothecat.catconomy.defaultImplementations;

import ninekothecat.catconomy.interfaces.ICurrencyPrefix;

@SuppressWarnings({"unused"})
public class CatPrefix implements ICurrencyPrefix {
    private final String shortPrefix;
    private final String longPrefix;

    public CatPrefix(String shortPrefix) {
        this.shortPrefix = shortPrefix;
        longPrefix = shortPrefix;
    }

    public CatPrefix(String shortPrefix, String longPrefix) {
        this.shortPrefix = shortPrefix;
        this.longPrefix = longPrefix;
    }

    @Override
    public String getShortPrefix() {
        return shortPrefix;
    }

    @Override
    public String getLongPrefix() {
        return longPrefix;
    }
}

File: src/main/kotlin/ninekothecat/catconomy/defaultImplementations/CatPrefix.kt
package ninekothecat.catconomy.defaultImplementations

import ninekothecat.catplugincore.money.interfaces.ICurrencyPrefix

class CatPrefix : ICurrencyPrefix {
    override val shortPrefix: String
    override val longPrefix: String

    constructor(shortPrefix: String) {
        this.shortPrefix = shortPrefix
        longPrefix = shortPrefix
    }

    constructor(shortPrefix: String, longPrefix: String) {
        this.shortPrefix = shortPrefix
        this.longPrefix = longPrefix
    }
}